http/digest/src/test/java/org/wildfly/security/http/digest/DigestMechanismFactoryTest.java

```diff
@@ -0,0 +1,83 @@
+import java.util.HashMap;
+
+public class DigestMechanismFactoryTest {
+
+    private DigestMechanismFactory digestMechanismFactory = new DigestMechanismFactory();
+
+    CallbackHandler dummyCallbackHandler = new CallbackHandler() {
+        @Override
+        public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+        }
+    };
+
+    @Test
+    public void testGetMechanismNamesPropertiesNull(){
+        digestMechanismFactory.getMechanismNames(null);
+        digestMechanismFactory.getMechanismNames(new HashMap<String,String>());
+    }
+
+    @Test
+    public void testGetMechanismNamesReturnNotNull(){
+        String[] mechanismNames = digestMechanismFactory.getMechanismNames(null);
+        Assert.assertNotNull("Array of mechanism names is not null.", mechanismNames);
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismMechanismNameNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism(null,new HashMap<String,String>(),dummyCallbackHandler);
+            Assert.fail("Mechanism name could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismPropertiesNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism("DIGEST",null,dummyCallbackHandler);
+            Assert.fail("Properties could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismCallbackHandlerNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism("DIGEST",new HashMap<String,String>(),null);
+            Assert.fail("CallbackHandler could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismBasicMechanismName() throws HttpAuthenticationException{
+        HttpServerAuthenticationMechanism httpServerAuthenticationMechanism = digestMechanismFactory.createAuthenticationMechanism("BASIC",new HashMap<String,String>(),dummyCallbackHandler);
+        Assert.assertNull("Provided mechanism must be null.", httpServerAuthenticationMechanism);
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismReturnNotNull() throws HttpAuthenticationException{
+        HttpServerAuthenticationMechanism httpServerAuthenticationMechanism = digestMechanismFactory.createAuthenticationMechanism("DIGEST",new HashMap<String,String>(),dummyCallbackHandler);
+        Assert.assertNotNull("HttpServerAuthenticationMechanism is not null.",httpServerAuthenticationMechanism);
```

Comment:
@keshav-redhat Just a detail. the message "HttpServerAuthenticationMechanism is not null." is being used to inform us what happened if the assertion failed. In this case, if the assertion fails it means that the `httpServerAuthenticationMechanism` is null. So please update to:

`"HttpServerAuthenticationMechanism is null."`

Reply:
@keshav-redhat This was not updated